### DISCUSSION TITLE
v4.3

### DISCUSSION THREAD

**Discussion by Tzahi12345** [Tzahi12345/YoutubeDL-Material#669] [Releases]
Hi everyone! It's been about a year and a half since the last update, so we've been due for one to say the least. If you're on the `nightly` tag, most of these you'll have already seen. If not, then you're in for a lot of improvements!

## Features
* MongoDB support!
  * MongoDB is now preferred over the default, local DB. It's much quicker and at scales, leads to a much smoother experience
  * You can transfer your local DB to MongoDB using the `Database` tab in the settings menu
  * No more corrupt DBs! Regardless, please back up your DB regularly
* New download manager
  * Downloads can now be tracked and managed through a new interface. Downloads can be paused, restarted, and errors can be viewed.
  * Subscriptions now use the new download manager, preventing subs from getting locked up checking for new downloads and streamlining the download process
* New tasks & maintenance page
  * Added ability to run or schedule specific tasks
    * Backup/Restore DB
      * MongoDB backups can be done through MongoDB, we simply store it into a JSON file locally
    * Check for DB records with missing files
    * Import missing DB records
    * Fix duplicate files in DB
    * Update youtube-dl (or whatever repo you use)
* Concurrent streams
  * You can now watch your videos simultaneously with friends! [Click here](https://github.com/Tzahi12345/YoutubeDL-Material/pull/378#issue-892631464) for a demo.
* File metadata can now be modified from the UI
  * Just click the info button on files and you'll see a new edit button on the top right of the dialog!
* Much faster downloads with yt-dlp as the new default downloader!

### UI

* Dedicated settings page
  * Settings dialog has been converted into a page for ease-of-use
  * Added option to rate-limit downloads
* Added pagination to files on the home screen
* Home screen files can now be filtered by type (audio or video)
* Version and commit info can now be viewed from the `About` dialog
* Updated subscription pages to use the new file cards (icons) from the home page
* Hovering over a file card will now show a preview of that video
* Fonts and bootstrap are now hosted locally (Thanks @firstdorsal for the PR!)
* Added ability to crop files prior to downloading
* New translations. We now support:
  * English
  * Spanish
  * Chinese
  * Dutch
  * French
  * German
  * Italian
  * Czech
  * Norwegian
  * Finnish
  * Korean
  * Macedonian
  * Portuguese
  * Russian
  * Telugu
  * Indonesian
  * Some of these are only partial translations. If you'd like to help out [click here!](https://hosted.weblate.org/projects/youtubedl-material/ytdl-material/). Special thanks to everyone who's helped out so far!

### Backend

* Added args overriding support
  * E.g. global custom args will override default args, per-subscription args will override all other args
* File sharing should work a lot more reliably
* Improved arg simulation (it now occurs on the backend)

### Dependencies/Integrations

* Added Kodi .nfo file generation support
* Helm chart added (Thanks @benashby for the PR!)
* Heroku deployments now use Docker
* Angular updated to v13
* PM2 is now used instead of nodemon to enable server restarts
* Updated Twitch VOD chat downloading to use new Twitch API
  * [Twitch-Chat-Downloader](https://github.com/PetterKraabol/Twitch-Chat-Downloader) is now required, and a client secret must be set along with a client ID
  * Benefit is chat downloads are much faster!
* [SponsorBlock](https://sponsor.ajay.app/) integration! Enable it in the settings and all supported videos will show a "skip ad" button where appropriate

### Behind-the-scenes

* API docs now automated
* Much improved wiki (Thanks to @GlassedSilver for the support!)
* Tons of new automation using GitHub Actions
* Added unit tests (coverage will be a long-term WIP)

## Bug fixes
* Way too many to name! Sorry if this section excites you, but we've worked hard to minimize bugs and make this an extra-stable release.

Thank you to everyone who has contributed since the last update. There are literally dozens of you but I can't express my appreciation enough. Even when I was too busy to work on the project, it brought me joy to see notifications of PRs going up and new translations submitted. A special thanks to @GlassedSilver for his work on community management, reviewing PRs, code improvements, and suggestions (as always).

We're now at 20 million Docker pulls which is an insane number. The Firefox browser extension alone has ~350 active concurrent users. There's still loads of improvements to come, but the last 5+ years has been incredibly fun. Happy downloading!

<hr /><em>This discussion was created from the release <a href='https://github.com/Tzahi12345/YoutubeDL-Material/releases/tag/v4.3'>v4.3</a>.</em>

**Comment by GlassedSilver** [comment]
This release really excites me! Love that everyone who had been on the latest tag until now gets to enjoy all the many lovely improvements that have been made in the past one and a half years. This will unite the user base into a much more coherent user experience again where there were those who experienced an outdated scope and experience of YTDL-M and those, who were in the know enough to see all that has happened all this time.

Good stuff!

**Comment by notDavid** [comment]
Am I correct in assuming it is advised to switch from the docker nightly build to latest now...? 🙂 🙏🏼

**Reply by GlassedSilver** [reply]
Yup! :)

**Comment by Coalbus** [comment]
Amazing stuff, the wait was 110% worth it. Genuinely impressed by everything you all are doing here.

**Comment by ibrahimk157** [comment]
Amazing job on this new release! I’m genuinely mind blown by the amount of improvements you’ve put into this! Can’t thank you enough for all of the time and effort you’ve put into this! Keep up the amazing work!

**Comment by rockbo842** [comment]
Thanks for all of your work on this!
